File: basics-data-structure/binarytree.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left, self.right = None, None


class Traverse(object):
    def __init__(self):
        self.traverse_path = list()

    def preorder(self, root):
        if root:
            self.traverse_path.append(root.value)
            self.preorder(root.left)
            self.preorder(root.right)

    def inorder(self, root):
        if root:
            self.inorder(root.left)
            self.traverse_path.append(root.value)
            self.inorder(root.right)

    def postorder(self, root):
        if root:
            self.postorder(root.left)
            self.postorder(root.right)
            self.traverse_path.append(root.value)

File: basics-data-structure/binarytree.js
class TreeNode {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

class Traverse {
    constructor() {
        this.traverse = [];
    }

    preorder(root) {
        if (root) {
            this.traverse.push(root.value);
            this.preorder(root.left);
            this.preorder(root.right);
        }
    }

    inorder(root) {
        if (root) {
            this.inorder(root.left);
            this.traverse.push(root.value);
            this.inorder(root.right);
        }
    }

    postorder(root) {
        if (root) {
            this.postorder(root.left);
            this.postorder(root.right);
            this.traverse.push(root.value);
        }
    }
}